lib/msf/core/post/linux/system.rb

```diff
@@ -96,6 +96,33 @@ def get_sysinfo
     return system_data
   end
 
+  #
+  # return true if possible chroot environment detect, if not true then false
+  #
+  def is_chroot?
+    chroot_env = false
+
+    if not directory?("/proc/")
+      chroot_env = true
+    else
+      dir("/proc/").each do |pid|
+        pid = Integer(pid) rescue nil
+        if pid
+          filename = ::File.readlink("/proc/#{pid}/exe") rescue nil
```

Comment:
will search for pid's out of chroot, if the /proc/[pid]/exe link not found, this may mean that you are in a chroot, because you dont have access to files out of chdir